OpenRA.Mods.Common/Scripting/Global/DateTimeGlobal.cs

```diff
@@ -20,11 +21,15 @@ namespace OpenRA.Mods.Common.Scripting
 	public class DateGlobal : ScriptGlobal
 	{
 		readonly TimeLimitManager tlm;
+		readonly int ticksPerSecond;
 
 		public DateGlobal(ScriptContext context)
 			: base(context)
 		{
 			tlm = context.World.WorldActor.TraitOrDefault<TimeLimitManager>();
+			var defaultGameSpeed = Game.ModData.Manifest.Get<GameSpeeds>().Speeds.FirstOrDefault(s => s.Key == "default");
+			var defaultTimestep = defaultGameSpeed.Value != null ? defaultGameSpeed.Value.Timestep : 40;
+			ticksPerSecond = 1000 / defaultTimestep;
```

Comment:
Why don't you use
```csharp
context.World.WorldActor.Trait<MapOptions>().GameSpeed.Timestep
```
?

Reply:
That would change behavior of all time-based triggers!
`ticksPerSecond` must be identical between game speeds.